test/strategies/latin_hypercube.jl

```diff
@@ -84,11 +84,54 @@ end
 
 @testset "Scale not a symbol" begin
 tree_model = DecisionTreeRegressor()
-forest_model = EnsembleMode(atom=tree_model)
+forest_model = EnsembleModel(atom=tree_model)
 r1 = range(forest_model, :(atom.n_subfeatures), lower=1, upper=9,
            scale = x->10^x)
 r2 = range(forest_model, :bagging_fraction, lower=0.4, upper=1.5);
 my_latin = LatinHypercube(nGenerations=2,popSize= 120)
 @test_throws ErrorException setup(tuning::LatinHypercube, model=forest_model,
                                   r=[r1,r2], verbosity)
 end
+
+@testset "Return value for ranges" begin
+
+    tree_model = DecisionTreeRegressor()
+    forest_model = EnsembleModel(atom=tree_model)
+    d = 2
+
+    r1 = range(forest_model, :(atom.n_subfeatures), lower=1., upper=9.);
+    r2 = range(forest_model, :bagging_fraction, lower=0.4, upper=1.5);
+    bounds, dims = _create_bounds_and_dims(d,[r1,r2])
+    @test all(bounds .== [(1,9),(0.4,1.5)]
+    @test all(dims .== [Continuous(),Continuous()])
+
+    r3 = range(forest_model, :(atom.n_subfeatures), lower=1., upper=9,
+                scale =:log);
+    r4 = range(forest_model, :bagging_fraction, lower=0.4, upper=1.5);
+    bounds, dims = _create_bounds_and_dims(d,[r3,r4])
+    @test all(bounds .== [(0.0, 2.1972245773362196),(0.4,1.5)]
```

Comment:
I love seeing all these tests!! Great job. 

Minor point, some of the testing is brittle. Instead of `==` use `≈` (or `approx`) when comparing floats. 

But this raises another issue. Since LatinSquare uses RNGs, there needs to be an interface point for the user to prescribe a RNG that is passed to the library function generating the "plan". It should be a hyperparameter of `LatinHypercube` of type `AbstractRNG` but the user should be able to specify an integer at construction to get `rng=MersenneTwister(user_specifed_rng)` - see [RandomSearch case](https://github.com/alan-turing-institute/MLJTuning.jl/blob/348a9f61b6958081f97c238a1b2517b9b758bb89/src/strategies/random_search.jl#L107) for the relevant logic. 

In testing however, one always passes `rng=StableRNGs.StableRNG(123)` or similar because the behaviour of the `MersenneTwister` and default `GLOBAL_RNG ` is not consistent over julia versions. Have a look at the `StableRNGs` package for some details. 

Let me know if a brief audio chat would be helpful to go over this stuff.

Reply:
Thanks for the constant encouragement, I feel I should have been faster on developing this, ahah. 

I see the problem of checking equality with floats, that was a bit dumb, fixed. 

I also see the RNG issue. I just have one problem: I cannot pass the rng to the library call, it uses a random permutation inside the source code, see https://github.com/MrUrq/LatinHypercubeSampling.jl/blob/master/src/LatinHypercubeSampling.jl line 101-110. I can randperm the plan output though following the rng input from the user. Is that what you have in mind? I don't think it helps though because there is still a random permutation from the library call though..

Reply:
https://github.com/MrUrq/LatinHypercubeSampling.jl/issues/15

Reply:
@ludoro Ask and you shall receive: https://github.com/MrUrq/LatinHypercubeSampling.jl/issues/15 .

Reply:
That's great, will add it.